app/src/main/java/org/gdg/frisbee/android/pulse/PulseAdapter.java

```diff
@@ -38,16 +39,20 @@
 
 class PulseAdapter extends BaseAdapter {
 
-    private LayoutInflater mInflater;
-    private ArrayList<Map.Entry<String, PulseEntry>> mPulse;
-    private int[] mPositions;
+    private final LayoutInflater mInflater;
+    private final ArrayList<Map.Entry<String, PulseEntry>> mPulse;
+    @Nullable
+    private final Directory mDirectory;
 
+    private int[] mPositions;
     private int mMode;
+    private boolean mCheckValuesAgainstDirectory;
 
-    public PulseAdapter(Context ctx, @Nullable int[] positions) {
+    public PulseAdapter(Context ctx, @Nullable int[] positions, Directory directory) {
```

Comment:
I think unfortunately the Nullable defined in field is not transitive to here. Should we add `@Nuallable` to the constructor param as well?